trace/otel_span.go

```diff
@@ -0,0 +1,110 @@
+package trace
+
+import (
+	"context"
+	"encoding/hex"
+	"time"
+
+	"go.opentelemetry.io/otel/api/kv"
+	oteltrace "go.opentelemetry.io/otel/api/trace"
+	"google.golang.org/grpc/codes"
+)
+
+// otelSpan implements otel/api/trace.Span. Though it defines methods to satisfy
+// that interface, none of them are implemented except for SpanContext. This is
+// used for compatibilty between beelines and OTel-Go.
+type otelSpan struct {
+	spanContext oteltrace.SpanContext
+}
+
+func (os otelSpan) SpanContext() oteltrace.SpanContext {
+	return os.spanContext
+}
+
+func (os otelSpan) IsRecording() bool {
+	return false
+}
+
+func (os otelSpan) SetStatus(code codes.Code, msg string) {
+	return
+}
+
+func (os otelSpan) SetAttribute(k string, v interface{}) {
+	return
+}
+
+func (os otelSpan) SetAttributes(attributes ...kv.KeyValue) {
+	return
+}
+
+func (os otelSpan) End(options ...oteltrace.EndOption) {
+	return
+}
+
+func (os otelSpan) RecordError(ctx context.Context, err error, opts ...oteltrace.ErrorOption) {
+	return
+}
+
+func (os otelSpan) Tracer() oteltrace.Tracer {
+	return nil
+}
+
+func (os otelSpan) AddEvent(ctx context.Context, name string, attrs ...kv.KeyValue) {
+	return
+}
+
+func (os otelSpan) AddEventWithTimestamp(ctx context.Context, timestamp time.Time, name string, attrs ...kv.KeyValue) {
+	return
+}
+
+func (os otelSpan) SetName(name string) {
+	return
+}
+
+// oTelRemoteSpanFromContext looks for an Otel SpanContext in the provided context.
+// If it finds a non-empty context, it returns it, otherwise it returns nil.
+func oTelRemoteSpanFromContext(ctx context.Context) *oteltrace.SpanContext {
+	spanContext := oteltrace.RemoteSpanContextFromContext(ctx)
+	emptyContext := oteltrace.SpanContext{}
```

Comment:
Is doing this on line line, `if oteltrace.SpanContext{} == spanContext` a more normal way of checking for a zero value struct in Go?

Reply:
Good catch. You can't have `{}` in a conditional, which is what threw me off originally, but there is an accessor which I can use. Updated.